### PR TITLE
feat(notifications): correct team fallback settings

### PR DESCRIPTION
**Pull request by scefali**
This PR changes the fallback logic if we can't send a notification to a team. This is needed because we are going to change the default settings for slack alerts to be on whereas previously it was off by default. Since we actually make a notification setting when you link a team (https://github.com/getsentry/sentry/blob/master/src/sentry/integrations/slack/views/link_team.py#L127-L136) it was never an issue before. If you had the notification setting for a team enabled, it meant there was an external actor row. But when we change the default that won't be the case anymore. As a result, we should check the `ExternalActor` row in addition to notification settings to see if we want to fallback to sending to the members. If there is no `ExternalActor` row for that provider, we will just fall back to sending to members individually as usual.

### CODE REVIEW

File: src/sentry/notifications/utils/participants.py [hunk 1]
```diff
@@ -402,3 +397,36 @@ def get_recipients_by_provider(
...
     return combine_recipients_by_provider(teams_by_provider, users_by_provider)
+
+
+def get_teams_by_provider(
+    project: Project, teams: Iterable[Team], notification_type: NotificationSettingTypes
+):
+    # figure out which teams have external actors for the right platform
+    team_actor_ids = list(map(lambda x: x.actor_id, teams))
+    external_actors = ExternalActor.objects.filter(
+        actor_id__in=team_actor_ids,
```
Comment (scefali) [inline src/sentry/notifications/utils/participants.py:408]: We have an index on this column so I think this query will be fast

**Comment by ceorourke** [comment]
If an organization doesn't have Slack installed, do they still have this default Slack team notification setting? They still have to link their team to a certain Slack channel so I'm confused how a default setting would work.

**Comment by scefali** [comment]
@ceorourke Yes, we still have Slack enabled by default in the UI. We could potentially query the Identity table and not show slack as enabled by default in the UI. Regardless, we show a notification in the UI saying that you don't have Slack enabled for a particular organization. Settings are just not tied to other state right now.

**Comment by ceorourke** [comment]
Which UI are you referring to here? The user settings one? 

Even if you do have Slack installed and a team Slack notification setting enabled, without the team linked to a channel there's no way for us to know where to send a notification, so I don't see why you'd enable the setting.

**Comment by scefali** [comment]
> setting

@ceorourke It's because if you have no settings stored at all, we want to default it to being on. Does that make sense?

**Pull request closed without merging**